src/mixins/scatter-mixin.js

```diff
@@ -57,20 +60,32 @@ export default function scatterMixin(_chart, _mapboxgl, mixinDraw = true) {
   function initializeXYDimsAndRanges(chart) {
     const xDims = []
     const yDims = []
+    const y2Dims = []
     const xRanges = []
     const yRanges = []
+    const y2Ranges = []
 
     addDimAndRange(chart.xDim(), xDims, xRanges)
     addDimAndRange(chart.yDim(), yDims, yRanges)
+    addDimAndRange(chart.y2Dim(), y2Dims, y2Ranges)
 
     if (typeof chart.getLayers === "function") {
       _chart.getLayers().forEach(layer => {
         if (typeof layer.xDim === "function") {
           addDimAndRange(layer.xDim(), xDims, xRanges)
         }
 
-        if (typeof layer.yDim === "function") {
-          addDimAndRange(layer.yDim(), yDims, yRanges)
+        if (
+          _chart.useTwoYAxes() &&
+          layer.layerType() === "crossSectionTerrain"
+        ) {
+          if (typeof layer.yDim === "function") {
```

Comment:
Could this `typeof layer.yDim === "function"` condition just be rolled up into the top `if`? It's checked separately in both the `if` and `else` it seems

Reply:
Definitely, good call

Reply:
Hm... now i'm second guessing this, if it's in the top level `if` then if it falls into the `else` it will mean it's not a function. Seems like this could be condensed somehow tho haha maybe an `if` wrapping the parent `if`... not necessary